WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -549,7 +553,21 @@ def update
         redirect_to root_url
       end
     elsif @competition.update(comp_params_minus_id)
-      if new_id && !@competition.update(id: new_id)
+      # The two extra updates here are done to automatically compute the cellname and ID for competitions with a short name.
+      m = VALID_NAME_RE.match(@competition.name) # These three lines are from models/competition.rb
+      name_without_year = m[1]
+      year = m[2]
+      year_space = " " + year
+      if @actually_using_organizer_view && @competition.name.length <= 32
+        @competition.update(cellName: name_without_year.truncate(MAX_CELL_NAME_LENGTH - year_space.length) + year_space)
+      end
+
+      if @actually_using_organizer_view && @competition.name.length <= 32
+        safe_name_without_year = ActiveSupport::Inflector.transliterate(name_without_year).gsub(/[^a-z0-9]+/i, '')
+        inferred_id = safe_name_without_year[0...(MAX_ID_LENGTH - year.length)] + year
+        @competition.update(id: inferred_id)
```

Comment:
There is already a section of code in `competition.rb`, method `create_id_and_cell_name`, which is automatically run on inserts and updates. Why is it not satisfactory to use/adapt that part of the code?

Reply:
I tried modifying the function you are referencing at first, however when I asked on the Slack I was told I could not get any pass to the function whether or not I am on the admin view page. Thus I moved it into this file. I agree this is annoying how the function is written basically two times across files, but I do not know how I can avoid that (given the subsequent updates depend on `@competition_admin_view`) .